clients/java/src/main/java/io/odpf/raccoon/client/RaccoonClient.java

```diff
@@ -0,0 +1,19 @@
+package io.odpf.raccoon.client;
+
+import io.odpf.raccoon.model.Event;
+import io.odpf.raccoon.model.Result;
+
+/**
+ * An interface for the raccoon clients.
+ */
+public interface RaccoonClient {
+    /**
+     * Sends a request to raccoon with the message provided.
+     *
+     * @param <T> The response type of the raccoon request.
+     *
+     * @param events The raccoon event message array.
+     * @return The raccoon response.
+     */
+    <T> Result<T> send(Event[] events);
```

Comment:
my only though was to intro this wrapper `Result<T>` type is, in the future if we introduce the v2 api and in that case if the response object changes then we don't have to change the signature .

Reply:
Technically, backward incompatible changes shouldn't be allowed since the Response object is an API contract not to be broken. If we do, then we will release a new major version that the client can choose to implement.